src/org/thoughtcrime/securesms/preferences/NotificationsPreferenceFragment.java

```diff
@@ -49,6 +49,7 @@ public void onCreate(Bundle paramBundle) {
     this.findPreference(Prefs.RINGTONE_PREF)
         .setOnPreferenceClickListener(preference -> {
           Uri current = Prefs.getNotificationRingtone(getContext());
+          if (current.toString().isEmpty()) current = null;  // silent
```

Comment:
this was the reason why if you set default notification sound to silent, when opening again no option was marked since empty Uri is not Silent, Silent == null